openwisp_users/apps.py

```diff
@@ -130,6 +137,22 @@ def connect_receivers(self):
             dispatch_uid='make_first_org_user_org_owner',
         )
 
+    @classmethod
+    def handle_organization_update(cls , instance , sender,**kwargs)  :
+        Organization = load_model("openwisp_users" , "Organization")
+        OrganizationUsers = load_model("openwisp_users" ,"OrganizationUser")
+        try:
+            old_obj = Organization.objects.get(pk=instance.pk)
+        except ObjectDoesNotExist:
+            logger.warning(f"Organization with pk {instance.pk} does not exist.")
+            return
+        except Exception as e:
+            logger.exception(f"An error occurred while fetching the organization: {e}")
+            return
+        if getattr(old_obj , "is_active") != getattr(instance , "is_active") :
+            for user in OrganizationUsers.objects.filter(organization = instance):
+                cls._invalidate_user_cache(user)
```

Comment:
If the database is big enough, this task can take enough time to cause the webserver to time out.
We need to move this logic to a background celery task.
We have a number of examples in other modules, eg: https://github.com/openwisp/openwisp-controller/blob/master/openwisp_controller/config/tasks.py#L75-L80
Ideally the celery task calls a class method in the model which takes care of the rest.